### ISSUE TITLE
Error in JW mapping with z2symmetry_reduction when running VQE-UCCSD

### ISSUE THREAD

**Issue by tanvipg** [Qiskit/qiskit-nature#241]
### Information

- **Qiskit Nature version**: qiskit-nature: 0.1.3
- **Python version**: Python 3.8.10 
- **Operating system**: macOS BigSur

### What is the current behavior?
When implementing VQE-UCCSD with JW Mapping with z2symmetry_reduction; HF state returns None object. Similar behavior observed for Parity Mapping with z2symmetry_reduction  and no two qubit reduction.

### Steps to reproduce the problem

from qiskit                                       import *
from qiskit_nature.drivers                        import PySCFDriver,UnitsType,HFMethodType,QMolecule
from qiskit_nature.problems.second_quantization   import ElectronicStructureProblem
from qiskit_nature.converters.second_quantization import QubitConverter
from qiskit_nature.mappers.second_quantization    import JordanWignerMapper,ParityMapper
from qiskit_nature.transformers                   import FreezeCoreTransformer
from qiskit_nature.circuit.library                import HartreeFock,UCCSD
from qiskit_nature.algorithms                     import VQEUCCFactory,GroundStateEigensolver
from qiskit.algorithms.optimizers                 import L_BFGS_B,COBYLA
from qiskit.algorithms                            import MinimumEigensolver,VQE
from qiskit.utils                                 import QuantumInstance
from qiskit.circuit.library                       import TwoLocal
from qiskit.opflow                                import PauliSumOp,TaperedPauliSumOp,Z2Symmetries,PauliOp

driver    = PySCFDriver(atom="Li 0 0 0; H 0 0 1.4",
                        unit=UnitsType.ANGSTROM,
                        charge=0,
                        spin=0,
                        basis='sto-3g',
                        hf_method=HFMethodType.RHF)

es_problem = ElectronicStructureProblem(driver,[FreezeCoreTransformer()]) 
second_q_op = es_problem.second_q_ops() 
num_particles = es_problem.num_particles
num_spin_orbitals = 2*es_problem.molecule_data_transformed.num_molecular_orbitals
print('Num particles:',num_particles)
print('Num molecular orbitals:',int(num_spin_orbitals/2))

qubit_conv_JWT  = QubitConverter(mapper=JordanWignerMapper(),two_qubit_reduction = False, z2symmetry_reduction = 'auto')
qubit_op_JWT = qubit_conv_JWT.convert(es_problem.second_q_ops()[0], num_particles = num_particles, sector_locator = es_problem.symmetry_sector_locator)

vqe_solver = VQEUCCFactory(quantum_instance=quantum_instance,
        optimizer= opt,
        initial_point= None,
        gradient = None,
        expectation = None,
        include_custom = False,
        ansatz = None, #default is UCCSD
        initial_state = None) 

calc = GroundStateEigensolver(qubit_conv_JWT, vqe_solver)
res = calc.solve(es_problem)
print(res)

### What is the expected behavior?

Tapered and transformed HF state should be returned, which is not None object

**Comment by Rukhsan** [comment]
Please assign this issue to me.

**Comment by mrossinek** [comment]
@Brogis1 was right. This does indeed appear to be fixed, too, now that we closed #342.

After fixing the above snippet by specifying a `quantum_instance` and `opt` (which were missing) I was successfully able to run the simulation without running into errors. Below is the output using `opt = None` (which falls back to `SLSQP` in Terra) and `quantum_instance = QuantumInstance(Aer.get_backend("aer_simulator_statevector"))` on the latest Terra+Nature:
```
=== GROUND STATE ENERGY ===

* Electronic ground state energy (Hartree): -9.012181741489
  - computed part:      -1.121390258355
  - FreezeCoreTransformer extracted energy part: -7.890791483133
~ Nuclear repulsion energy (Hartree): 1.133951166257
> Total ground state energy (Hartree): -7.878230575231

=== MEASURED OBSERVABLES ===

  0:  # Particles: 2.000 S: 0.000 S^2: 0.000 M: 0.000

=== DIPOLE MOMENTS ===

~ Nuclear dipole moment (a.u.): [0.0  0.0  2.64561657]

  0:
  * Electronic dipole moment (a.u.): [0.0  0.0  4.47392393]
    - computed part:      [0.0  0.0  4.47710745]
    - FreezeCoreTransformer extracted energy part: [0.0  0.0  -0.00318352]
  > Dipole moment (a.u.): [0.0  0.0  -1.82830736]  Total: 1.82830736
                 (debye): [0.0  0.0  -4.64709334]  Total: 4.64709334
```
The qubit operator obtained for this problem with the JW mapping and automatic Z2 symmetry reduction required 6 qubits.
The result above is numerically identical to that obtained with the full 10-qubit operator when no symmetry reductions are used.

Therefore, I am closing this issue :+1:
